File: src/main/kotlin/com/itmo/microservices/demo/order/api/dto/CatalogItemDto.java
package com.itmo.microservices.demo.order.api.dto;

import com.itmo.microservices.demo.order.impl.entity.CatalogItemEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.context.annotation.Lazy;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Lazy
public class CatalogItemDto {
    private UUID uuid;
    private String title;
    private String description;
    private int price;
    private int amount;

    public CatalogItemEntity toEntity() {
        return new CatalogItemEntity(this.uuid, this.title, this.description, this.price, this.amount);
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/warehouse/api/model/CatalogItemDto.kt
package com.itmo.microservices.demo.warehouse.api.model

import java.util.*
import javax.annotation.Nullable
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

data class CatalogItemDto(
    var id: UUID,
    @field:NotNull
    val title: String,
    @field:NotNull
    val description: String,
    @field:NotNull
    @field:Min(value = 1, message = "Price must be greater than 0")
    val price: Int,
    val amount: Int
)
